src/Avalonia.Base/Rendering/Composition/CompositingRenderer.cs

```diff
@@ -266,12 +269,18 @@ private void UpdateCore()
                 renderTransform *= (-offset) * visual.RenderTransform.Value * (offset);
             }
 
-
-
             comp.TransformMatrix = MatrixUtils.ToMatrix4x4(renderTransform);
 
             _recorder.BeginUpdate(comp.DrawList);
+
+            var oldRenderOptions = _recordingContext.RenderOptions;
```

Comment:
Render options aren't getting recorded into drawing operations, so it's pointless to have them in the DrawingContext.

Reply:
We need this to be able to create GlyphRun implementations with the current render options